api/util/GenerateTokenScript.js

```diff
@@ -0,0 +1,104 @@
+const fs = require('fs');
+const readline = require('readline');
+const { google } = require('googleapis');
+
+/**
+ * Prompts user for client ID & secret and writes them to config
+ * file containing auth credentials, then generates a token.json file for GCP
+ * @param {String} configPath the directory path of the config.json file
+ */
+const setAuthCredentials = async (configPath) => {
+  const rawConfig = fs.readFileSync(configPath);
+  console.debug('Welcome to SCE\'s Google Cloud API Token Generator!');
+
+  if(rawConfig) {
+    const clientID = await inputPrompt('Please enter the Client ID: ');
+    const clientSecret = await inputPrompt('Please enter the Client Secret: ');
+
+    const config = JSON.parse(rawConfig);
+    if(clientID) {
+      config.googleApiKeys.CLIENT_ID = clientID;
+    }
+    if(clientSecret) {
+      config.googleApiKeys.CLIENT_SECRET = clientSecret;
+    }
+
+    fs.writeFile(configPath, JSON.stringify(config), (err) => {
+      if(err) return console.debug('ERR', err);
+      console.debug('Successfully written to:', configPath);
+
+      generateToken(
+        config,
+        ['https://mail.google.com/'],
+        __dirname + '/../config/token.json',
+      );
+    });
+
+  } else {
+    console.debug('Config file DNE at path:', configPath);
+  }
+};
+
+/**
+ * Generates an auth token to make API calls, assuming configData exists
+ * @param {Object} configData data from config file containing auth credentials
+ * @param {String | String[]} apiScopes which API(s) to interact with
+ * @param {String} tokenPath path that the auth token file will be written to
+ */
+const generateToken = async (configData, apiScopes, tokenPath) => {
+  const oauth2Client = new google.auth.OAuth2(
+    configData.googleApiKeys.CLIENT_ID,
+    configData.googleApiKeys.CLIENT_SECRET,
+    configData.googleApiKeys.REDIRECT_URIS[0],
+  );
+
+  const authUrl = oauth2Client.generateAuthUrl({
+    /* eslint-disable-next-line camelcase */
+    access_type: 'offline',
+    scope: apiScopes,
+  });
+  console.debug('\nAuthorize this app by visiting this URL:', authUrl);
+
+  const authCode = await inputPrompt(
+    '\nPlease enter the \'Authorization Code\' from the URL here: '
+  );
+  rl.close();
+
+  if(authCode) {
+    oauth2Client.getToken(authCode, (err, token) => {
+      if(err) console.debug('Error generating token', err);
+      oauth2Client.setCredentials(token);
+
+      console.debug(`\nGenerating token.js file to ${tokenPath}...`);
+      fs.writeFileSync(tokenPath, JSON.stringify(token));
+      console.debug('Done! You are ready to use GCP!');
+    });
+
+  } else {
+    console.debug('Error with authorization code');
+  }
+};
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+/**
+ * Prompts user for input and resolves the input or false
+ * @param {String} prompt question to get the info we need
+ */
+const inputPrompt = (prompt) => {
```

Comment:
nice, good idea abstracting this, i'd put this at the top of the file to improve readability